File: Codechef/APRIL21/kavgmat.cpp
#include<bits/stdc++.h>
using namespace std;
#define ll long long
#define se second
#define loop(i,a,b) for(ll i = a ; i <= b ; i++)
#define endl "\n"
#define vi vector<ll>
#define fastio ios_base::sync_with_stdio(false);cin.tie(NULL)
#define w(x)  ll x; cin>>x; while(x--)

void solve() {

    ll n,m,k,ans = 0;
    cin >> n >> m >> k;
    vector<vi> psa(n+1, vi(m+1));
    loop(i,0,n) psa[i][0] = 0;
    loop(i,0,m) psa[0][i] = 0;

    loop(i,1,n) {
    	loop(j,1,m) {
    		cin >> psa[i][j];
    		ans += (psa[i][j] >= k);
    		psa[i][j] += psa[i-1][j] + psa[i][j-1] - psa[i-1][j-1];
    	}
    }

    loop(i,1,n) {
    	loop(j,1,m) {
    		ll l = 1;
    		while(i+l <= n and j+l <= m) {
    			ll x = i+l, y = j+l;
    			ll num = (l+1)*(l+1);
    			ll total = psa[x][y] - psa[x][j-1] - psa[i-1][y] + psa[i-1][j-1];
    			if(total*1.0/num >= k) ans += 1;
    			l += 1;
    		}
    	}
    }

    cout << ans << "\n";
}


int main()
{
	fastio;

    #ifndef ONLINE_JUDGE
        freopen("/home/devang/input.txt","r",stdin);
        freopen("/home/devang/output.txt","w",stdout);
    #endif

    w(tc)
    	solve();
	
	return 0;
}

File: Codechef/APRIL21/kavgmat.py
def solve():
	n,m,k = map(int, input().split())
	a = [ [ 0 for _ in range(m+1) ] for _ in range(n+1) ]
	for i in range(1,n+1):
		a[i] = [0] + list(map(int, input().split()))
	
	ans = 0

	for i in range(1,n+1):
		for j in range(1,m+1):
			ans += 1 if a[i][j] >= k else 0
			a[i][j] = a[i-1][j] + a[i][j-1] - a[i-1][j-1] + a[i][j]

	for i in range(1,n+1):
		for j in range(1,m+1):
			l = 1
			x, y = i+l, j+l

			while x <= n and y <= m:
				num = (l+1)*(l+1)
				total = a[x][y] - a[x][j-1] - a[i-1][y] + a[i-1][j-1]
				avg = total/num
				if avg >= k:
					ans += 1
				l += 1
				x, y = i+l, j+l


	print(ans) 


tc = int(input())
for _ in range(tc):
	solve()
			
